browser/ui/views/brave_javascript_tab_modal_dialog_view_views.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_BRAVE_JAVASCRIPT_TAB_MODAL_DIALOG_VIEW_VIEWS_H_
+#define BRAVE_BROWSER_UI_VIEWS_BRAVE_JAVASCRIPT_TAB_MODAL_DIALOG_VIEW_VIEWS_H_
+
+#include "base/memory/weak_ptr.h"
+#include "chrome/browser/ui/views/javascript_tab_modal_dialog_view_views.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/views/widget/widget_observer.h"
+
+namespace web_modal {
+class WebContentsModalDialogHost;
+}  // namespace web_modal
+
+class BraveJavaScriptTabModalDialogViewViews
+    : public JavaScriptTabModalDialogViewViews,
+      public views::WidgetObserver {
+  METADATA_HEADER(BraveJavaScriptTabModalDialogViewViews,
+                  JavaScriptTabModalDialogViewViews)
+ public:
+  BraveJavaScriptTabModalDialogViewViews(
+      content::WebContents* parent_web_contents,
+      content::WebContents* alerting_web_contents,
+      const std::u16string& title,
+      content::JavaScriptDialogType dialog_type,
+      const std::u16string& message_text,
+      const std::u16string& default_prompt_text,
+      content::JavaScriptDialogManager::DialogClosedCallback dialog_callback,
+      base::OnceClosure dialog_force_closed_callback);
+  ~BraveJavaScriptTabModalDialogViewViews() override;
+
+ private:
+  friend class JavaScriptTabModalDialogManagerDelegateDesktop;
+
+  web_modal::WebContentsModalDialogHost* GetModalDialogHost();
```

Comment:
This function will never return `nullptr` by the looks of the implementation, therefore it is better to have it returning a reference, i.e. `web_modal::WebContentsModalDialogHost&`.